src/createStore.test.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const createStore = require('./createStore');
+const ACTION_INCREASE = 'increase';
+const ACTION_ADD = 'add';
+
+const stateWithCountReducer = (state, action) => {
+  const { type } = action;
+  switch (type) {
+    case ACTION_INCREASE:
+      return {
+        ...state,
+        count: state.count + 1,
+      };
+
+    case ACTION_ADD:
+      return {
+        ...state,
+        count: state.count + action.value,
+      };
+
+    default:
+      return state;
+  }
+};
+
+{
+  const initialState = { count: 0 };
+  const store = createStore(stateWithCountReducer, initialState);
+
+  store.dispatch({ type: 'increase' });
+  store.dispatch({ type: 'increase' });
+  store.dispatch({ type: 'increase' });
+
+  test('store should return count equal to 3', () => {
+    expect(initialState).toMatchObject({ count: 0 });
+    expect(store.getState()).toMatchObject({ count: 3 });
+  });
```

Comment:
please do not test the `reducer` just check if it receives `state` and `action` as arguments with https://jestjs.io/docs/en/mock-functions ,that its result is set as a new state and callback added with `subscribe` are called once after each dispatch